desktopIconsIntegration.js

```diff
@@ -0,0 +1,160 @@
+            if (extension.state === ExtensionUtils.ExtensionState.ENABLED) {
+                this._sendMarginsToExtension(extension);
+                return;
+            }
+            // if the extension is being disabled, we must do a full refresh, because if there were other extensions originally
+            // loaded after that extension, those extensions will be disabled and enabled again without notification
+            this._changedMargins();
+        });
+    }
+
+    /**
+     * Sets or updates the top, bottom, left and right margins for a
+     * monitor. Values are measured from the monitor border (and NOT from
+     * the workspace border).
+     *
+     * @param {int} monitor Monitor number to which set the margins.
+     *                      A negative value means "the primary monitor".
+     * @param {int} top Top margin in pixels
+     * @param {int} bottom Bottom margin in pixels
+     * @param {int} left Left margin in pixels
+     * @param {int} right Right margin in pixels
+     */
+    setMargins(monitor, top, bottom, left, right) {
+        this._margins[monitor] = {
+            'top': top,
+            'bottom': bottom,
+            'left': left,
+            'right': right
+        };
+        this._changedMargins();
+    }
+
+    /**
+     * Clears the current margins. Must be called before configuring the monitors
+     * margins with setMargins().
+     */
+    resetMargins() {
+        this._margins = {};
+        this._changedMargins();
+    }
+
+    /**
+     * Disconnects all the signals and removes the margins.
+     */
+    destroy() {
+        if (this._emID) {
+            this._extensionManager.disconnect(this._emID);
+            this._emID = 0;
+        }
+        if (this._timedMarginsID) {
+            GLib.source_remove(this._timedMarginsID);
+            this._timedMarginsID = 0;
+        }
+        this._margins = null;
+        this._changedMargins();
+    }
+
+    _changedMargins() {
+        if (this._timedMarginsID) {
+            GLib.source_remove(this._timedMarginsID);
+        }
+        this._timedMarginsID = GLib.timeout_add(GLib.PRIORITY_DEFAULT, 100, ()=> {
+            this._sendMarginsToAll();
+            this._timedMarginsID = 0;
+            return GLib.SOURCE_REMOVE;
+        });
+    }
+
+    _sendMarginsToAll() {
+        for (let uuid of this._extensionManager.getUuids()) {
+            let extension = this._extensionManager.lookup(uuid);
+            this._sendMarginsToExtension(extension);
+        }
+    }
+
+    _sendMarginsToExtension(extension) {
+        // check that the extension is an extension that has the logic to accept
+        // working margins
+        if ((extension?.state !== ExtensionUtils.ExtensionState.ENABLED) ||
+            (extension?.stateObj?.DesktopIconsUsableArea?.uuid !== IdentifierUUID)) {
+                return;
+        }
+        extension.stateObj.DesktopIconsUsableArea.setMarginsForExtension(Me.uuid, this._margins);
```

Comment:
I'd like this style better :stuck_out_tongue: 

```suggestion
        if (extension?.state !== ExtensionUtils.ExtensionState.ENABLED))
            return;

        const usableArea = extension?.stateObj?.DesktopIconsUsableArea;
         if (usableArea?.uuid === INDENTIFIER_UUID))
                usableArea.setMarginsForExtension(Me.uuid, this._margins);
```